scalalib/src/mill/javalib/android/AndroidAppModule.scala

```diff
@@ -0,0 +1,276 @@
+   *
+   * This method:
+   *
+   * - Takes the unsigned APK and optimizes it for better performance on Android devices.
+   *
+   * @return A `PathRef` to the aligned APK file.
+   *
+   * @see [[createApk]]
+   */
+  def alignApk: T[PathRef] = T {
+    val alignedApk =
+      T.dest / s"${appName().toString}.aligned.apk" // Specify output aligned APK file name.
+
+    os.call(
+      Seq(
+        zipalignPath().path.toString, // Path to the zipalign tool
+        "-f",
+        "-p",
+        "4", // Force overwrite and align with a page size of 4
+        createApk().path.toString, // Use the unsigned APK from createApk
+        alignedApk.toString // Specify output aligned APK file
+      )
+    )
+
+    PathRef(alignedApk) // Return the path to the aligned APK.
+  }
+
+  /**
+   * Step 6: Sign the APK using a keystore.
+   *
+   * This method:
+   *
+   * - Signs the aligned APK with a keystore. If the keystore does not exist, it generates one.
+   *
+   * @return A `PathRef` to the signed APK file.
+   *
+   * @see [[alignApk]]
+   * @see [[createKeystore]]
+   */
+  def createApp: T[PathRef] = T {
+    val signedApk =
+      projectRoot() / s"${appName().toString}.apk" // Specify output signed APK file name.
+
+    os.call(
+      Seq(
+        apksignerPath().path.toString,
+        "sign", // Command to sign APK
+        "--ks",
+        createKeystore().path.toString, // Use the keystore from createKeystore
+        "--ks-key-alias",
+        "androidkey", // Alias for the key
+        "--ks-pass",
+        "pass:android", // Keystore password
+        "--key-pass",
+        "pass:android", // Key password
+        "--out",
+        signedApk.toString, // Specify output signed APK file
+        alignApk().path.toString // Use the aligned APK from alignApk
+      )
+    )
+
+    PathRef(signedApk) // Return the path to the signed APK.
+  }
+
+  /**
+   * Creates a keystore for signing APKs if it doesn't already exist.
+   *
+   * This method:
+   *
+   * - Generates a keystore file using the `keytool` command.
+   *
+   * @return A `PathRef` to the keystore file.
+   *
+   * @see [[createApp]]
+   */
+  def createKeystore: T[PathRef] = T {
+    val keystoreFile = T.dest / "keystore.jks" // Specify keystore file name.
+
+    if (!os.exists(keystoreFile)) {
+      os.call(
+        Seq(
+          "keytool",
+          "-genkeypair",
+          "-keystore",
+          keystoreFile.toString, // Generate keystore
+          "-alias",
+          "androidkey", // Key alias
+          "-dname",
+          "CN=MILL, OU=MILL, O=MILL, L=MILL, S=MILL, C=IN", // Distinguished name
+          "-validity",
+          "10000", // Validity period in days
+          "-keyalg",
+          "RSA", // Key algorithm
+          "-keysize",
+          "2048", // Key size
+          "-storepass",
+          "android", // Keystore password
+          "-keypass",
+          "android" // Key password
+        )
+      )
+    }
```

Comment:
Every task shelling out to a third-party binary should contain a link to the docs for the third party binary where you got the command from

Reply:
done